src/you_get/extractors/netease.py

```diff
@@ -106,18 +107,41 @@ def netease_video_download(vinfo, output_dir='.', info_only=False):
                             output_dir=output_dir, info_only=info_only)
 
 def netease_song_download(song, output_dir='.', info_only=False, playlist_prefix=""):
+    def _download():
+        netease_download_common(title, url_best,
+                                output_dir=output_dir, info_only=info_only)
+
     title = "%s%s. %s" % (playlist_prefix, song['position'], song['name'])
     songNet = 'p' + song['mp3Url'].split('/')[2][1:]
 
-    if 'hMusic' in song and song['hMusic'] != None:
-        url_best = make_url(songNet, song['hMusic']['dfsId'])
-    elif 'mp3Url' in song:
-        url_best = song['mp3Url']
-    elif 'bMusic' in song:
-        url_best = make_url(songNet, song['bMusic']['dfsId'])
+    k = 0
+    while k < 3:
```

Comment:
What's the point of using a while loop?

The logic here is to skip `netease_download_common` when `url_best` is not valid. So why not
```py
if url_best is None:
    netease_download_common(title, url_best, output_dir=output_dir, info_only=info_only)
```
or even
```py
try:
    netease_download_common(title, url_best, output_dir=output_dir, info_only=info_only)
```